src/main/java/seedu/address/model/recipe/RecipeContainsIngredientsPredicate.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.recipe;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Recipe}'s {@code Ingredient} matches any of the ingredients available in the inventory.
+ */
+public class RecipeContainsIngredientsPredicate implements Predicate<Recipe> {
```

Comment:
Since this predicate is for inventory ingredients, should we call it `RecipeContainsInventoryIngredientsPredicate` which accepts the actual `ReadOnlyInventory` in the constructor? Then we will be able directly call `getIngredientList()` on it and use streams to obtain the names. In this case, we would not need the additional methods in `Inventory` and `ReadOnlyInventory`. Now, it seems like the abstraction of the ingredient names is being broken.

Reply:
Okay, this can be done. However I need Recipe#getIngredientNameString to get a string of the ingredients required in a recipe, does that break the abstraction?

Reply:
would `Ingredient#isSameIngredient()` be useful here? it only calls `IngredientName#equals()` which ignores the case in string comparison.

Reply:
Good idea. I have implemented it using streams, I'll PR it.